site/_posts/2021-01-14-softozor-fusionauth.md

```diff
@@ -0,0 +1,98 @@
+
+**Dan:** What problems did we solve for you? And how were you solving them before FusionAuth?
+
+**Laurent:** You solve the permission management problem. You provide us with a reliable way to generate JWTs that are compatible with hasura. It's the first time where we had to integrate an authorization and authentication system. 
+
+> Both [the .NET and python SDKs] saved us a lot of time. 
+
+**Dan:** Why did you choose FusionAuth over the competition?
+
+**Laurent:** We evaluated a few options: keycloak, keratin/authn, and FusionAuth. The ease of installation and the set of features were the most decisive motivations to use your software. Its documentation is outstanding, and that also helped a lot in choosing FusionAuth. 
+
+**Dan:** How much time and money would you say FusionAuth has saved you?
+
+**Laurent:** It saved a lot of time. If we agree that time is money, then it saved us a lot of money too. We could've implemented a reduced set of the features provided by FusionAuth, but that would've taken a lot of time. We could've chosen another solution than FusionAuth and implemented the missing features, but again that would've been a killer. I'd say it saved us something like one man-year of time. 
+
+**Dan:** How do you run FusionAuth (k8s, standalone tomcat server, behind a proxy, etc)?
+
+**Laurent:** We run FusionAuth on a k8s cluster. It is completely hidden behind our hasura api. We write hasura actions that interact with FusionAuth.  
+
+> I'd say FusionAuth saved us something like one man-year of time.
+
+**Dan:** Can you tell me a bit more about the hasura actions? 
+
+**Laurent:**  Our system is based on the 3-factor app pattern: realtime graphql API, reliable eventing, and async serverless. That is out-of-the-box hasura behavior. 
+
+Whenever some relevant data are changed in our database, some events are triggered, which results in calling some serverless functions (e.g. in our system we use OpenFaaS). In addition to that, we can complete the GraphQL API automatically generated by hasura with additional queries / mutations, which are known as "hasura actions". For example, the automatically generated API does not come out with a login mutation. Either you route the login request from your frontend to FusionAuth directly or you decide that you have a central API over which everything should go. 
+
+We chose the latter variant, as we find it clearer. In that case, we had to define a hasura login action. That amounts to telling hasura what the mutation looks like: what name, what input, what output, and then link that mutation with a serverless function. In that context, we are very happy to use the .Net SDK provided by FusionAuth! It has all the necessary REST client wrappers necessary for us to perform the operations we need on FusionAuth. 
```

Comment:
In the first sentence, I'd say "as we find it clearer and cleaner" or something like that. It is both clear and clean.